eng/pipelines/templates/jobs/smoke-tests.yml

```diff
@@ -0,0 +1,135 @@
+parameters:
+  - name: Daily
+    default: true
+  - name: Artifacts
+    type: object
+    default: []
+
+jobs:
+  - ${{ if eq(parameters.Daily, false) }}:
+    - job: smoke_test_eligibility
+      displayName: Check Smoke Test Eligibility
+      steps:
+        # pipelines only supports template each when it generates the entire script block.
+        - ${{ each artifact in parameters.Artifacts }}:
+          - ${{if and(ne(variables['Skip.Release'], 'true'), ne(artifact.skipPublishPackage, 'true')) }}:
+            - pwsh: |
+                $deps = (Select-Xml -Path ./common/SmokeTests/SmokeTest/SmokeTest.csproj `
+                                    -Xpath //ItemGroup/PackageReference/@Include).Node.Value
+                if ($deps.Contains("${{ artifact.name }}")) {
+                  Write-Host "Smoke tests will run for ${{ artifact.name }}"
+                  Write-Host "##vso[task.setvariable variable=RunSmokeTests]true"
+                } else {
+                  Write-Host "Smoke tests will NOT run for ${{ parameters.package }}"
+                }
+              name: check_smoke_tests_${{ artifact.safeName }}
+              displayName: Check smoke test eligibility for ${{ artifact.name }}
+        - pwsh: |
+            Write-Host "Setting RunSmokeTests to $($env:RunSmokeTests)"
```

Comment:
Tip: for namespace-scoped variables, just `"... ${env:RunSmokeTests}"` will work. It's also slightly faster (not that it matters here; just a tip) because it doesn't create a nested script AST to parse and execute.

Reply:
...same below.